src/subscript/ri_wellmod/ri_wellmod.py

```diff
@@ -0,0 +1,586 @@
+"""
+ ri_wellmod.py
+
+
+"""
+import argparse
+from pathlib import Path
+
+import fnmatch
+import shutil
+import xml.dom.minidom
+import re
+import logging
+from datetime import datetime
+
+import rips
+import grpc
+from subscript import getLogger
+
+DESCRIPTION = """
+``ri_wellmod`` is a utility to generate Eclipse well model definitions
+(WELSPECS/WELSPECL, COMPDAT/COMPDATL, WELSEGS, COMPSEGS) using ResInsight. The script
+takes as input a ResInsight project with wells and completions defined, in addition to
+an Eclipse case (either an initialized case or an input case with grid and PERMX|Y|Z
+and NTG defined in the GRDECL format).
+
+.. note:: Well names specified as command line arguments are assumed to refer to the
+   Eclipse well names, i.e., the completion export names as defined in the ResInsight
+   wells project.
+"""
+
+CATEGORY = "modelling.reservoir"
+
+EXAMPLES = """
+.. code-blck:: console
+
+ FORWARD_MODEL RI_WELLMOD(
+    <RI_PROJECT>=<CONFIG_PATH>/../../resinsight/input/well_modelling/wells.rsp,
+    <ECLBASE>=<ECLBASE>,
+    <OUTPUTFILE>=<RUNPATH>/eclipse/include/schedule/well_def.sch)
+
+ FORWARD_MODEL RI_WELLMOD(
+    <RI_PROJECT>=<CONFIG_PATH>/../../resinsight/input/well_modelling/wells.rsp,
+    <ECLBASE>=<ECLBASE>,
+    <OUTPUTFILE>=<RUNPATH>/eclipse/include/schedule/well_def.sch,
+    <MSW>="A2;A4;'R*')
+
+ FORWARD_MODEL RI_WELLMOD(
+    <RI_PROJECT>=<CONFIG_PATH>/../../resinsight/input/well_modelling/wells.rsp,
+    <ECLBASE>=<ECLBASE>,
+    <OUTPUTFILE>=<RUNPATH>/eclipse/include/schedule/well_def.sch,
+    <MSW>="A4",
+    <XARG0>="--lgr",
+    <XARG1>="A4:3;3;1")
+
+
+.. warning:: Remember to remove line breaks in argument list if copying the examples
+   into your own ERT config.
+
+
+.. note:: More examples and options may be seen in the subscript docs for the script
+   ``ri_wellmod``, just replace ',' by ';' and note that spaces cannot be part of
+   argument strings, so you may need to use <XARGn> for the individual parts.
+
+"""
+
+
+logger = getLogger(__name__)
+
+RI_HOME = "/prog/ResInsight"
+DEFAULT_VERSION = "2020.10.1"
+
+
+class CustomFormatter(
+    argparse.ArgumentDefaultsHelpFormatter, argparse.RawDescriptionHelpFormatter
+):
+    """
+    Multiple inheritance used for argparse to get both
+    defaults and raw description formatter
+    """
+
+    # pylint: disable=unnecessary-pass
+    pass
+
+
+def get_resinsight_exe(version=DEFAULT_VERSION):
+    """
+    Return the path to a valid ResInsight install, False if not found.
+    """
+    ri_exe = shutil.which("ResInsight")
+    if ri_exe is not None:
+        return ri_exe
+
+    ri_exe = Path(RI_HOME + "/resinsight_" + version + "_RHEL7/ResInsight")
+    if ri_exe.exists():
+        return ri_exe
+
+    return False
+
+
+def get_parser():
+    """
+    Utility function to build the cmdline argument parser using argparse
+    (https://docs.python.org/3/library/argparse.html)
+    """
+
+    description = """
+  Script description.
```

Comment:
Remove, or write something more interesting :) Avoid the indent in the string.

Reply:
Updated (cookiecutter leftover...)